src/core/src/sys2/interactions.rs

```diff
@@ -0,0 +1,369 @@
+// Lumol, an extensible molecular simulation engine
+// Copyright (C) 2015-2016 Lumol's contributors — BSD license
+
+//! The `Interactions` type for storing particles types to potentials
+//! associations.
+
+use std::f64;
+use std::collections::BTreeMap;
+use std::cmp::{min, max};
+
+use energy::{PairInteraction, BondPotential, AnglePotential, DihedralPotential};
+use energy::{GlobalPotential, CoulombicPotential};
+use sys::ParticleKind as Kind;
+
+/// Sort pair indexes to get a canonical representation
+#[inline] fn normalize_pair(i: Kind, j: Kind) -> (Kind, Kind) {
```

Comment:
I was just wondering, since in every comment where these functions appear, we talk about "sorting".